### DISCUSSION TITLE
Scaling the Stack: Vertical vs Horizontal

### DISCUSSION THREAD

**Discussion by elmau** [mozilla/hubs#3283] [Hubs Cloud Questions] [answered]
I would like to know what's the difference between scaling vertically vs horizontally.

What's the benefit for choosing one over the other, and in which scenario one is better than the other.

**Comment by robinkwilson** [comment, marked as answer]
**TL;DR** 
Scale vertically (increase instance size) before horizontal (adding machines to an Enterprise multi-server setup). 
Use a 2 app x 2 stream server + scale up as necessary (allow 1 failure, with another to take its place). For EXTRA large instances >10,000 CCU or more, a 4x4 setup can work.

**Longer Explanation**
Best scenario is scaling vertically before scaling horizontally. For a beefy setup, a 2x2 (2 app x 2 stream) should be sufficient. For REALLY large setups (more than 10,000 CCU expected), you can do a 4x4 or 8x8 setup. 2x2 because if one machine goes down unexpectedly, there's another to take its place and your users will not notice a difference.

The bottleneck with horizontal scaling is Let's Encrypt certificates (if I'm remembering correctly). For every EC2 instance you spin up, we automatically generate certificates for these machines and you can run out of free certs if you scale horizontally too far. From what I remember some folks scaled to 12x12 and ran out of certificates so the system ran into issues.
